### PR TITLE
[autoscaler] Create provider exactly once

### LINKED ISSUE #10692: [autoscaler] monitor.py has a memory/thread leak
### What is the problem?

With this cluster yaml, you can watch monitor.py via `htop` quickly eat up all of the available memory.

It also seems to be creating hundreds of NodeUpdater threads.

*Ray version and other system information (Python version, TensorFlow version, OS):* master

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):

```
cluster_name: memoryleak
min_workers: 2
max_workers: 2
initial_workers: 2
autoscaling_mode: default
target_utilization_fraction: 0.8
idle_timeout_minutes: 60
provider:
  type: aws
  region: us-east-1
  cache_stopped_nodes: false
auth:
  ssh_user: ubuntu
head_node:
  BlockDeviceMappings:
  - DeviceName: /dev/sda1
    Ebs:
      VolumeSize: 300
      DeleteOnTermination: true
      VolumeType: gp2
  ImageId: latest_dlami
  InstanceType: t3.xlarge
worker_nodes:
  BlockDeviceMappings:
  - DeviceName: /dev/sda1
    Ebs:
      VolumeSize: 300
      DeleteOnTermination: true
      VolumeType: gp2
  ImageId: latest_dlami
  InstanceType: t3.xlarge
file_mounts: {}
cluster_synced_files:
- ~/my-cluster-synced-file
file_mounts_sync_continuously: true
initialization_commands: []
setup_commands:
- pip install --user -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp37-cp37m-manylinux1_x86_64.whl
head_setup_commands:
- tmux new-session -d "python3 -m http.server"
- pip install --user -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp37-cp37m-manylinux1_x86_64.whl
- sleep 5
- curl localhost:8000 -o ~/my-cluster-synced-file
worker_setup_commands: []
```
cc @ericl @wuisawesome

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

 - session_* is misleading because it prints output from prior sessions
 - avoids creating the provider 100 times, closes #10692 

@wuisawesome
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/autoscaler/updater.py [hunk 1]
```diff
@@ -204,7 +206,10 @@ def do_sync(remote_path, local_path, allow_non_existing_paths=False):
             with cli_logger.group(
                     "Processing worker file mounts",
                     _numbered=("[]", previous_steps + 2, total_steps)):
+                cli_logger.print("synced files: {}",
+                                 str(self.cluster_synced_files))
                 for path in self.cluster_synced_files:
+                    path = os.path.expanduser(path)
```
Comment (wuisawesome) [inline python/ray/autoscaler/updater.py:212]: Can we pull this out into a separate PR and do it in the intializer (keep it consistent with the `self.file_mounts` treatment?

File: python/ray/autoscaler/autoscaler.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ class StandardAutoscaler:
     StandardAutoscaler is also used to bootstrap clusters (by adding workers
     until the target cluster size is met).
     """
+    provider = None
```
Comment (wuisawesome) [inline python/ray/autoscaler/autoscaler.py:57]: Please do this in the constructor so we don't have a random class variable (and it's easier to track/see)

File: python/ray/autoscaler/commands.py [hunk 3]
```diff
@@ -455,7 +455,8 @@ def kill_node(config_file, yes, hard, override_cluster_name):
 
 def monitor_cluster(cluster_config_file, num_lines, override_cluster_name):
     """Tails the autoscaler logs of a Ray cluster."""
-    cmd = "tail -n {} -f /tmp/ray/session_*/logs/monitor*".format(num_lines)
+    cmd = "tail -n {} -f /tmp/ray/session_latest/logs/monitor*".format(
+        num_lines)
```
Comment (ijrsvt) [inline python/ray/autoscaler/commands.py:459]: can we f-string this?

**Review by wuisawesome — APPROVED** [review]

**Review by ijrsvt — APPROVED** [review]
LGTM! But where does this close #10702 ?

File: python/ray/autoscaler/commands.py (hunk 3 above)
Comment (wuisawesome) [reply python/ray/autoscaler/commands.py:459]: i just moved the #10702 part to a different pr see #10705

**Comment by wuisawesome** [comment]
> LGTM! But where does this close #10702 ?

i just moved the #10702 part to a different pr see #10705

File: python/ray/autoscaler/autoscaler.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/autoscaler/autoscaler.py]: thanks
